### ISSUE TITLE
Wrong results for some numbers close to zero

### ISSUE THREAD

**Issue by potocpav** [mikedilger/float-cmp#2]
The `ulps` function seems to return wrong results in this test-case. I don't know much about f64 representation, so the best thing I can do about it is to report it :-)

``` rust

extern crate float_cmp;

use std::mem;
use float_cmp::Ulps;

fn main() {
    let (ai64, bi64): (i64, i64) = (4364452196894661639, 4364352217599516529);
    let (a, b): (f64, f64) = unsafe {
        (mem::transmute(ai64), mem::transmute(bi64))
    };
    println!("(a, b) == ({:.19}, {:.19})", a, b);
    // should be small numbers?
    println!("    a.ulps(&b) == {:20}", a.ulps(&b));
    println!(" (0.).ulps(&a) == {:20}", (0.0).ulps(&a));
    // overflow
    println!("(-0.).ulps(&a) == {:20}", (-0.0).ulps(&a));
}

/* Output:

(a, b) == (0.0000000000000000612, 0.0000000000000000600)
    a.ulps(&b) ==       99979295145110
 (0.).ulps(&a) == -4364452196894661639
thread '<main>' panicked at 'arithmetic operation overflowed', /home/pavel/.cargo/git/checkouts/float-cmp-2988329a307f998d/master/src/lib.rs:122

*/
```

**Comment by mikedilger** [comment]
They shouldn't be small numbers.   Your two i64s are quite far apart, considering they differ at the 5th digit.   And even though these two floats look close:
  0.0000000000000000612,
  0.0000000000000000600
They are not close in ULPs terms, in the same way that 612 != 600, and aparently there are 99979295145110 floating point representations between these two  (Floating point representations get denser and denser as you get closer to zero)

As for the overflow, this is something that they added to rust after I wrote this library.  I've just pushed a commit using wrapping_sub() to avoid the overflow.
